homeassistant/components/nextbus/coordinator.py

```diff
@@ -52,6 +52,10 @@ async def _async_update_data(self) -> dict[str, Any]:
         """Fetch data from NextBus."""
         self.logger.debug("Updating data from API. Routes: %s", str(self._route_stops))
 
+        if self.hass.state is not CoreState.running:
```

Comment:
It seems quite wierd that this integration would be dependent on HA startup (are we fixing a symptom?).

However it would be better to use `from homeassistant.helpers.start import async_at_started` in `__init__.py` to delay the startup of this integration rather than returning some empty dict here. See example in `speedtestdotnet`.

Reply:
I updated the PR to use this pattern instead. I verified that this resolves the issue on my production HA instance.